dfsclient/src/main/java/org/hpccsystems/dfs/client/HPCCFile.java

```diff
@@ -262,6 +262,15 @@ public FileFilter getFilter()
     public HPCCFile setFilter(String filterexpression) throws Exception
     {
         this.filter = new FileFilter(filterexpression);
+
```

Comment:
do the other setters setProjectList, setClusterRemapInfo, setTargetfilecluster, setFileAccessExpirySecs need to update private members?

Reply:
Nope, this was the only setter that affected the DataPartition

Reply:
We do have some HpccFile filter unit tests on the Spark side, but they didn't hit the ordering issue that the DataSource exposed. It is probably a good idea to move those over to the DFSClient unit tests and expand them a bit. I will create a JIRA

Reply:
Although I'd like to get rid of the "clusterremapper", it seems to affect the parts:
ClusterRemapper clusterremapper = ClusterRemapper.makeMapper(clusterRemapInfo, fileinfoforread);
this.dataParts = DataPartition.createPartitions(fileinfoforread.getFileParts(), clusterremapper,0, filter, fileinfoforread.getFileAccessInfoBlob(), fileType);

Perhaps the remapper cannot be set once the fileparts have already been created, same thing for the setTargetfilecluster, and setFileAccessExpirySecs

Reply:
You are right clusterremapper would affect things, I missed that when I first looked at it. 
Cluster remapper complicates things, because it is destructive in that we won't have the original IPs after it does its job. I will look into the viability of not generating the data partitions until they are explicitly asked for. That would still potentially leave us with an issue for clusterremapper changes, I guess it can default to recreating the data partitions or throwing an error.

Reply:
@jpmcmu those setters should error out if the parts have already been created